### PR TITLE
Adding pact-message readme and refactoring the existing one

### PR DESCRIPTION
**Pull request by neilcampbell**
@nirpet Readme updates to include the details of how to use pact-message.

### CODE REVIEW

**Comment by nirpet** [comment]
@neilcampbell Cool, thanks for that. 
Will try to go over it this weekend.

File: docs/pact-message.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+Pact will invoke your handler with the expected message, so you can validate your handler functions as desired.
+
+#### 1. Build your message consumer
+In order to make a message consumer testable with Pact, we need to isolate the message queue/protocol/infrastructure bindings (message listener) from the message handling code (message handler).  
+Pact only needs to validate the message handlers and therefore this readme doesn't list any message listener code.
+
+A message handler might look like below:
+
+```c#
+public class SavePets
```
Comment (nirpet) [inline docs/pact-message.md]: Rename to SavePetsHandler?

File: Samples/MessageZooEvents/ZooEventsConsumer/Program.cs [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
         {
-            //TODO: It would be nice to show a Rabbit/AMQP example?
+            var handler = new SavePets(new PetRepo());
 
-            //You're messaging/event technology binding/wiring would go here
-            //which would to register a handler for events that are published
+            //This is the message listener, who's sole purpose is to receive the message from the queue/stream/infrastructure and forward onto the correct message handler.
+            //You're messaging/event technology binding/wiring would go here.
+
+            //Received from the Queue. It would be nice to show a Rabbit/AMQP example?
```
Comment (nirpet) [inline Samples/MessageZooEvents/ZooEventsConsumer/Program.cs]: Put under TODO and one of us will implement it in another pull request

File: Samples/MessageZooEvents/ZooEventsProducer/Program.cs [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
-            //TODO: It would be nice to show a Rabbit/AMQP example?
+            //This is the message sender, who's sole purpose is to call the message generator and send the generated messages to the queue/stream/infrastructure.
+            //You're messaging/event technology binding/wiring would go here.
+
+            var animalCreator = new AnimalCreator();
 
-            //You're messaging/event technology binding/wiring would go here
-            //which would publish events
+            var message = animalCreator.CreateAPet("Fred", PetType.Fish);
+            //Send to the queue. It would be nice to show a Rabbit/AMQP example?
```
Comment (nirpet) [inline Samples/MessageZooEvents/ZooEventsProducer/Program.cs]: Put under TODO and one of us will implement it in another pull request

File: docs/pact-message.md (hunk 1 above)
Comment (neilcampbell) [reply docs/pact-message.md]: I think I will leave it as SavePets as I'm describing what the thing does rather than the message it handles.

**Pull request merged**